src/main/java/com/google/googleidentity/oauth2/util/OAuth2Utils.java

```diff
@@ -47,10 +48,30 @@
 
     }
 
+    /**
+     * @return whether the uri matches one of the uri in uriList
+     */
+    public static boolean matchUri(List<String> uriList, String uri) {
+        for (String eachPattern : uriList) {
+            if (uri.startsWith(eachPattern)) {
+                if (eachPattern.equals(uri)) {
+                    return true;
+                }
+                if (eachPattern.charAt(eachPattern.length()-1) == '/') {
```

Comment:
Add comments explaining the rationale for each of these cases. Consider giving examples of a pattern and input uri that would match due to each case.